src/test/scala/nodes/learning/ScalaGMMSuite.scala

```diff
@@ -0,0 +1,102 @@
+package nodes.learning
+
+import breeze.linalg._
+import org.apache.spark.SparkContext
+import org.scalatest.FunSuite
+import pipelines._
+import utils.{MatrixUtils, Stats}
+
+class ScalaGMMSuite extends FunSuite with LocalSparkContext with Logging {
+
+  test("GMM Single Center") {
+    sc = new SparkContext("local", "test")
+
+    val k = 1
+
+    val data = sc.parallelize(Array(
+      DenseVector[Double](1.0, 2.0, 6.0),
+      DenseVector[Double](1.0, 3.0, 0.0),
+      DenseVector[Double](1.0, 4.0, 6.0)
+    ))
+
+    val center = DenseVector[Double](1.0, 3.0, 4.0).asDenseMatrix
+    val gmm = ScalaGMMEstimator(k, minClusterSize = 1).fit(data)
+
+    assert(gmm.means.t === center)
+
+    // TODO: TEST Variances!
+    gmm.apply(data)
+  }
+
+  test("GMM Two Centers") {
```

Comment:
One thing to do here is actually generate the data as a mixture of Gaussians with a known mean/variance structure and then see if the model we recover is close to the data-generating model.